text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,384 @@
+cases, large enterprises might create their own intermediary telemetry schemas
+for custom definitions.
+
+Each signal definition defined in the Component Telemetry Schema, where possible,
+reuses the existing syntax and semantics defined by the semantic conventions.
+Each signal definition is also identified by a unique name (or ID), making
+schemas addressable, easy to traverse, validate, and diff.
+
+The key design principles to be followed in the definition of the Resolved
+Telemetry Schema are:
+
+* **Self-contained**: No external references are allowed. This artifact contains
+  everything required to determine what an application or a library produces in
+  terms of telemetry.
+* **Easy to exchange**: This artifact must be easily accessible from a web
+  server via a URL. This artifact must be small and avoid the repetition of
+  definitions.
+* **Easy to parse**: A widespread and well-defined format should be preferred.
+  JSON is an example of such a format.
+* **Easy to interpret**: The internal structure of this artifact must be
+  straightforward to avoid any misinterpretation and must be efficient.
+* **Platform- and Language-agnostic**: This artifact must be independent of any
+  platform architectures and programming languages.
+
+The following diagram describes two main use cases for the Resolved Telemetry
+Schema. The key points to remember are: 1) both use cases result in a Resolved
+Telemetry Schema, 2) Resolved Telemetry Schemas serve as the mechanism for
+distributing Telemetry Schemas throughout the entire ecosystem, and 3) Resolved
+Telemetry Schemas would replace/augment existing SchemaURL.
+
+![Use cases](./img/0240-otel-weaver-use-cases.svg)
+
+Note: The relationship between Telemetry Schema v1.1
+([OTEP 0152](https://github.com/open-telemetry/oteps/blob/main/text/0152-telemetry-schemas.md))
+and the Component and Resolved Telemetry Schema concepts is still being
+discussed. This will be clarified in future OTEPs (refer to the last section).
+
+The following diagram illustrates a possible instance of a complex hierarchy of
+schemas and semantic convention registries. It involves several vendor and
+enterprise artifacts, in addition to the standardized OpenTelemetry artifacts.
+The schema resolution process will produce a self-contained Resolved Telemetry
+Schema that can be easily consumed by various tools and applications, such as a
+Client SDK generator, compatibility checker, compliance checker, data catalog
+feeder, and more.
+
+![Example of Telemetry Schema Hierarchy](./img/0240-otel-weaver-hierarchy.svg)
+
+This hierarchy of telemetry schemas helps large organizations in
+collaborating on the Component Telemetry Schema. It enables different
+aspects of a Component Telemetry Schema to be managed by various teams.
+
+For all the elements that make up the Component Telemetry Schema, a
+general mechanism of annotation or tagging will be integrated in order to
```

Comment:
> annotation or tagging

I understand this is vague on purpose, but one thing I'd want to avoid is having two independent DSLs, one for schema definition and another for annotations. If you noticed in Meta's schema-first telemetry paper, the annotations and the schema both used the exact same Thrift IDL mechanism (which, in turn, means annotations can be arbitrarily nested).

Reply:
Yes we are on the same page. I will clarify this part of the OTEP.

Reply:
Added the following sentence: This annotation mechanism will be included as part of the Component Telemetry Schema definition.